File: scripts/makeAdmin.js
"use strict";

// Donne (ou retire) les droits admin à un utilisateur.
// Usage : npm run make-admin -- <email> [--off]

const { sequelize, User } = require("../src/models");

async function run() {
  const args = process.argv.slice(2);
  const email = (args.find((a) => !a.startsWith("--")) || "").trim().toLowerCase();
  const on = !args.includes("--off");
  if (!email) {
    console.error("Usage : npm run make-admin -- <email> [--off]");
    process.exit(1);
  }

  await sequelize.authenticate();
  await sequelize.sync();
  const user = await User.findOne({ where: { email } });
  if (!user) {
    console.log(`✗ Aucun utilisateur avec l'email « ${email} ».`);
    await sequelize.close();
    return;
  }
  user.is_admin = on;
  await user.save();
  console.log(`✓ ${email} : is_admin = ${on}`);
  await sequelize.close();
}

run().catch((e) => {
  console.error("✗ Échec :", e);
  process.exit(1);
});

File: scripts/make_admin.py
"""Donne (ou retire) les droits admin à un utilisateur.

Usage :  python scripts/make_admin.py <email> [--off]
"""
import sys
import os
import argparse

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))

from paname import create_app
from paname.extensions import db
from paname.models import User


def run(email, on=True):
    app = create_app()
    with app.app_context():
        user = User.query.filter_by(email=email.strip().lower()).first()
        if not user:
            print(f"✗ Aucun utilisateur avec l'email {email!r}.")
            return
        user.is_admin = on
        db.session.commit()
        print(f"✓ {email} : is_admin = {on}")


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("email")
    parser.add_argument("--off", action="store_true", help="retirer les droits admin")
    args = parser.parse_args()
    run(args.email, on=not args.off)
